src/data/abilities/apply-ab-attrs.ts

```diff
@@ -31,53 +55,58 @@ export function getAbApplyFunc(mode: AbilityApplyMode) {
     case AbilityApplyMode.REVEALED:
       return applyRevealedAbAttrs;
     case AbilityApplyMode.IGNORE:
-      return () => [];
+      return ignoreAbAttrs;
   }
 }
 
 //#endregion
-//#region Helpers
+//#region Internal Functions
 
 /**
  * Applies a Pokemon's ability attributes of matching type
+ * @template TAttr The specific ability attribute type.
```

Comment:
```suggestion
 * @typeParam TAttr - The specific ability attribute type.
```
should also add the `-` in the `@param`s

Reply:
Id' prefer to stick to `@template`. My internal tsdoc highlighter can connect `TAttr` then. If I use `@typeParam` (which is from the official tsdoc though), it doesn't work.

Reply:
the `@template`s were replaced by `@typeParam`s in #1216 because they raise warnings due to invalid syntax for tsdoc as you said.
If we want to use `template` we should decide this as a group, add it as a custom tag in the tsdoc config and replace them all back. We don't want to have a mix
I'd rather we stick to typeParam for now and make a different PR if we decide to use `@template` instead